### PR TITLE
PR for #4009: crash in VR plugin

### PR DESCRIPTION
**Pull request by edreamleo**
See #4009.

- [x] Fix the crash in VR plugin.
- [x] Fix the corresponding crash in the VR3 plugin.

**Extras**

- [x] Add jedi to requirements.

### CODE REVIEW

**Comment by edreamleo** [comment]
@tbpassin @gatesphere I am going to merge this right away.

Please feel free to comment. I can always make changes later.

**Review by tbpassin — COMMENTED** [review]
1. I thought that deleteLater() was needed to delete Qt C++ stuff.  When will this happen without it?

2.  The Jedi requirement has nothing to do with the PR.  In the future, please don't mix unrelated changes in one PR like this.

**Comment by edreamleo** [comment]
@tbpassin Thanks for your comments.

1. Experiments show that closing an outline suffices to delete all widgets, including the VR pane.
  I'm not sure why I ever thought deleteLater was needed.
  Otoh, the "starting from scratch" protocol will have to take care that everything gets deleted as desired.
2. You're right, of course, but sometimes I don't think a little bit extra is so bad when I'm going to merge a PR quickly.

**Comment by tbpassin** [comment]
Will you *PLEASE* merge my pending changes for VR3 before you go changing anything else.  I already had to recreate them once because they were overlooked:

https://github.com/leo-editor/leo-editor/pull/3984

**Comment by edreamleo** [comment]
@tbpassin Done.

In the future _PLEASE_ remember to assign the current milestone to PRs. Otherwise I may never know they are pending.

**Comment by tbpassin** [comment]
Oops, I thought I had done that.

That reminds me, sometimes we get issues or PRs that don't have a label or milestone assigned.  Sometimes the submitter doesn't realize they should be.  But I think that ordinary users cannot add labels or milestones for themselves.  When I notice new ones I try to remember to make assignments.  But it's easy to forget.

I'm not sure how to resolve this.  If everyone can assign or change labels and milestones there could be chaos.  If they can't, their submission can get overlooked.

**Comment by edreamleo** [comment]
@tbpassin I think we devs just have to remember :-)

File: leo/plugins/viewrendered3.py [hunk 1]
```diff
@@ -1558,7 +1558,7 @@ def getVr3(event):
         controllers[h] = vr3 = viewrendered(event)
     return vr3
 #@+node:TomP.20191215195433.16: ** vr3.Commands
-#@+node:TomP.20191215195433.18: *3* g.command('vr3') (**weird**)
+#@+node:TomP.20191215195433.18: *3* g.command('vr3') (**weird effect**)
```
Comment (gatesphere) [inline leo/plugins/viewrendered3.py:1561]: Curious what 'weird effect' means here.

**Review by gatesphere — COMMENTED** [review]
General thoughts line up with @tbpassin, but the end result is that this appears to prevent the repeating error in log & console on both platforms I'm able to test on.

If closing an outline truly destroys all widgets, then probably good to go.

Agreed that mixing unrelated things into a PR is generally not a best practice.

Thanks for the fix!

**Comment by gatesphere** [comment]
> @tbpassin I think we devs just have to remember :-)

Remember what -- to occasionally scan through the list of issues without milestone tags?

I'm hesitant to apply milestones because I don't know @edreamleo's master plan for any given release.  As a result, some of my issues linger for a bit before they're noticed.  I suppose I could get over that and just be overzealous with them, and let Edward sort it out on his end.

**Comment by edreamleo** [comment]
@gatesphere Hi Jake. Thanks for your comments.

I'll try to remember to scan for PRs without milestones. I wouldn't worry about assigning milestones yourself. I often change them.

**Comment by edreamleo** [comment]
@gatesphere The weird effect is that on my Windows 11 machine enabling the VR3 plugin brings up the GE Force experience window!!! I have no idea how that could possibly happen.

**Comment by gatesphere** [comment]
> @gatesphere The weird effect is that on my Windows 11 machine enabling the VR3 plugin brings up the GE Force experience window!!! I have no idea how that could possibly happen.

Pardon the expletive, but wtf?  That's a new one by me...   Then again, after this week, I'll believe anything.

**Comment by edreamleo** [comment]
@gatesphere Whiskey Tango Foxtrot indeed.

**Pull request merged**